src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs

```diff
@@ -295,31 +292,41 @@ private static bool HasProjectFileChanges(ProjectChanges changes)
                    changes.GetRemovedAnalyzerReferences().Any();
         }
 
+        private BuildHostProcessManager? _applyChangesBuildHostProcessManager;
 
-        private IProjectFile? _applyChangesProjectFile;
-#endif
+        /// <summary>
+        /// The loaded project file that we are currently applying changes to. This is set in <see cref="ApplyProjectChanges(ProjectChanges)"/> if we're modifying a project that is going to require
+        /// file changes; it's cleared once we're done with that project.
+        /// </summary>
+        private IRemoteProjectFile? _applyChangesProjectFile;
```

Comment:
nit: move to top of file.

Reply:
the comment could use work.  what does it mean to be "done with that project".  

since many projects could need changing, what ensures this is serialized properly and we dont' stomp on things?

Reply:
TryApplyChanges loops through each project and applies changes one at a time. This approach is no different than how it worked before.